sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -364,7 +370,7 @@ def assign_rows_csr(X,
     """Densify selected rows of a CSR matrix into a preallocated array.
 
     Like out[out_rows] = X[X_rows].toarray() but without copying.
-    Only supported for dtype=np.float64.
+    No-copy supported for both dtype=np.float32 and dtype=np.float64.
```

Comment:
Ah you are right.
Thanks for pointing out my big mistake!
That should be in [this pr](https://github.com/scikit-learn/scikit-learn/pull/6588).